### PR TITLE
Skip the build step in the Realm import job if using a custom image

### LINKED ISSUE #11575: Realm import using operator should not run build if a custom image is provided
### Describe the bug

The Realm Import Job always runs a `kc.sh build` command before importing.
When the user provides a custom image the build step should be skipped.

### Version

18.0.0

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by andreaTP**
Resolves #11575

### CODE REVIEW

File: operator/src/test/java/org/keycloak/operator/RealmImportE2EIT.java [hunk 1]
```diff
@@ -25,6 +26,10 @@ public class RealmImportE2EIT extends ClusterOperatorTest {
     @Test
     public void testWorkingRealmImport() {
         // Arrange
+        // Recreating the database and the realm import CR to keep this test isolated
+        k8sclient.load(getClass().getResourceAsStream("/example-realm.yaml")).inNamespace(namespace).delete();
+        deleteDB();
+        deployDB();
```
Comment (vmuzikar) [inline operator/src/test/java/org/keycloak/operator/RealmImportE2EIT.java:32]: It seems we should isolate each test here. Maybe we could move this to a `@BeforeEach` method?

File: operator/src/test/java/org/keycloak/operator/RealmImportE2EIT.java [hunk 2]
```diff
@@ -67,6 +72,38 @@ public void testWorkingRealmImport() {
...
+        var keycloak = getDefaultKeycloakDeployment();
+        keycloak.getSpec().setImage(customImage);
+        deployKeycloak(k8sclient, keycloak, false);
+
+        // Act
+        k8sclient.load(getClass().getResourceAsStream("/example-realm.yaml")).inNamespace(namespace).createOrReplace();
+
+        // Assert
```
Comment (vmuzikar) [inline operator/src/test/java/org/keycloak/operator/RealmImportE2EIT.java:90]: Should we maybe add a similar assert to this to check that build is not present?
https://github.com/keycloak/keycloak/blob/5289b8704de36e6bfc9edcecffa7b639052465cb/operator/src/test/java/org/keycloak/operator/KeycloakDeploymentE2EIT.java#L344-L345
WDYT?

**Review by vmuzikar — CHANGES_REQUESTED** [review]

File: operator/src/test/java/org/keycloak/operator/RealmImportE2EIT.java (hunk 2 above)
Comment (DGuhr) [reply operator/src/test/java/org/keycloak/operator/RealmImportE2EIT.java:90]: also wanted to ask if this tests really checks if the build does not take place when a custom image is given. as i can see it it is just checking if the import takes place, but not if the build actually happens or not, right? so this test would also be green when a build takes place as of now without the additional assert.

File: operator/src/test/java/org/keycloak/operator/RealmImportE2EIT.java (hunk 1 above)
Comment (andreaTP) [reply operator/src/test/java/org/keycloak/operator/RealmImportE2EIT.java:32]: not all, but most, ok I can refactor 👍

File: operator/src/test/java/org/keycloak/operator/RealmImportE2EIT.java (hunk 2 above)
Comment (andreaTP) [reply operator/src/test/java/org/keycloak/operator/RealmImportE2EIT.java:90]: something on those lines 👍  ok

**Comment by andreaTP** [comment]
@vmuzikar @DGuhr ready for next round 💪  🙂 

Note:
improving the test case hit #11767

**Review by vmuzikar — DISMISSED** [review]
@andreaTP LGTM, thanks!

**Review by abstractj — DISMISSED** [review]
LGTM, merging it after @DGuhr approval.

**Comment by andreaTP** [comment]
@abstractj rebased on the latest `main` to fix the merge conflict.

This PR has been approved for 17 days now, if the tests are green I don't see any outstanding objection to merging it.

**Review by abstractj — APPROVED** [review]
@andreaTP thanks, waiting for the GHA checks to pass.

**Review by DGuhr — APPROVED** [review]
LGTM now, and sorry for the delayed reply :)

**Pull request merged**